pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -124,21 +130,27 @@
                 nativeQueries = new ArrayList<NativeQuery>();
               }
 
-              nativeQueries.add(new NativeQuery(nativeSql.toString(), toIntArray(bindPositions),isCurrentReWriteCompatible));
+              nativeQueries.add(new NativeQuery(nativeSql.toString(),
+                  toIntArray(bindPositions), DMLCommand.createStatementTypeInfo(
+                  current, isBatchedReWriteConfigured, isCurrentReWriteCompatible,
+                  isReturningPresent, isAutoCommit, nativeQueries.size())));
             }
             // Prepare for next query
             if (bindPositions != null) {
               bindPositions.clear();
             }
             nativeSql.setLength(0);
+            current = DMLCommandType.BLANK;
+            isReturningPresent = false;
           }
           break;
 
         case 'i':
           if (Parser.parseInsertKeyword(aChars, i)) {
```

Comment:
correct me if I'm wrong but the code for 'i' and 'I' look identical. For readability and avoiding duplication can we do 
case 'i':
case 'I':

same for 'r' and 'R'